gt/main.c

```diff
@@ -131,36 +129,78 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 		info->inner_ip_ver = ETHER_TYPE_IPv4;
 		info->l4_proto = inner_ipv4_hdr->next_proto_id;
 		info->l4_hdr = &inner_ipv4_hdr[1];
+
+		if (rte_ipv4_frag_pkt_is_fragmented(inner_ipv4_hdr)) {
+			info->frag = true;
+			info->l2_len = parsed_len;
```

Comment:
At this point doesn't `parsed_len` include the length of the L3 header too?

Reply:
Yes, the ``parsed_len`` includes the length of the outer L3 header.

Reply:
OK, it just stands out to me because we're assigning both the L2 and outer L3 lengths to something called `l2_len`, but we're doing this as a sort of trick, right? Because we need the fragmentation reassembly algorithm to know how far to skip over to get to the appropriate IP header?

If that's the case, maybe we could rename `info->l2_len` and `info->l3_len` to `info->l2_outer_l3_len` and `info->inner_l3_len` or something like that? To make it more clear what we're doing.